services/venice-controller/src/main/java/com/linkedin/venice/controller/systemstore/SystemStoreRepairTask.java

```diff
@@ -0,0 +1,236 @@
+package com.linkedin.venice.controller.systemstore;
+
+import com.linkedin.venice.common.VeniceSystemStoreType;
+import com.linkedin.venice.common.VeniceSystemStoreUtils;
+import com.linkedin.venice.controller.Admin;
+import com.linkedin.venice.controller.UserSystemStoreLifeCycleHelper;
+import com.linkedin.venice.controller.VeniceParentHelixAdmin;
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.pushmonitor.ExecutionStatus;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class SystemStoreRepairTask implements Runnable {
+  public static final Logger LOGGER = LogManager.getLogger(SystemStoreRepairTask.class);
+  public static final String SYSTEM_STORE_REPAIR_JOB_PREFIX = "CONTROLLER_SYSTEM_STORE_REPAIR_JOB_";
+
+  private final int pushStatusPollIntervalInSeconds = 30;
+  private final int heartbeatCheckIntervalMs = 100;
+
+  private final int heartbeatWaitTimeSeconds;
+  private final int maxRepairRetry;
+  private final VeniceParentHelixAdmin parentAdmin;
+  private final AtomicLong badMetaStoreCount = new AtomicLong(0);
+  private final AtomicLong badPushStatusStoreCount = new AtomicLong(0);
+  private final AtomicBoolean isRunning;
+
+  public SystemStoreRepairTask(
+      VeniceParentHelixAdmin parentAdmin,
+      int maxRepairRetry,
+      int heartbeatWaitTimeSeconds,
+      AtomicBoolean isRunning) {
+    this.parentAdmin = parentAdmin;
+    this.maxRepairRetry = maxRepairRetry;
+    this.heartbeatWaitTimeSeconds = heartbeatWaitTimeSeconds;
+    this.isRunning = isRunning;
+  }
+
+  @Override
+  public void run() {
+    for (String clusterName: parentAdmin.getClustersLeaderOf()) {
```

Comment:
@gaojieliu And here we are iterating through cluster. Even though we will check if it is the leader controller in the internal task, I still think it will be great to separate them into one task per cluster. What do you think?

Reply:
StoreBackupVersionCleanupService but I took another look at other service in controller. Looks like it is OK to for loop over cluster.